junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/VintageDiscoverer.java

```diff
@@ -56,31 +47,24 @@ public TestDescriptor discover(EngineDiscoveryRequest discoveryRequest, UniqueId
 	}
 
 	private TestClassCollector collectTestClasses(EngineDiscoveryRequest discoveryRequest) {
-		Predicate<Class<?>> classFilter = createTestClassPredicate(discoveryRequest);
+		ClassFilter classFilter = ClassFilter.of(buildClassNamePredicate(discoveryRequest), isPotentialJUnit4TestClass);
 		TestClassCollector collector = new TestClassCollector();
-		for (DiscoverySelectorResolver selectorResolver : getAllDiscoverySelectorResolvers(discoveryRequest)) {
+		for (DiscoverySelectorResolver selectorResolver : getAllDiscoverySelectorResolvers()) {
 			selectorResolver.resolve(discoveryRequest, classFilter, collector);
 		}
 		return collector;
 	}
 
-	private List<DiscoverySelectorResolver> getAllDiscoverySelectorResolvers(EngineDiscoveryRequest request) {
-		Predicate<String> classNamePredicate = buildClassNamePredicate(request);
-		return asList( //
-			new ClasspathRootSelectorResolver(classNamePredicate), //
-			new PackageNameSelectorResolver(classNamePredicate), //
-			new ClassSelectorResolver(), //
-			new MethodSelectorResolver(), //
-			new UniqueIdSelectorResolver()//
+	private List<DiscoverySelectorResolver> getAllDiscoverySelectorResolvers() {
+		return asList(
+		// @formatter:off
+			new ClasspathRootSelectorResolver(),
+			new PackageNameSelectorResolver(),
+			new ClassSelectorResolver(),
+			new MethodSelectorResolver(),
+			new UniqueIdSelectorResolver()
+		// @formatter:on
```

Comment:
Since these are now all stateless, we could change the list into a constant, right?

Reply:
Sure. Stored in final immutable field?

Reply:
> Since these are now all stateless, we could change the list into a constant, right?

Good catch!

> Sure. Stored in final immutable field?

Yep

Reply:
Where is `List.of(...)` when you need it?! ;)